lldb/source/Core/ValueObject.cpp

```diff
@@ -363,22 +363,24 @@ CompilerType ValueObject::MaybeCalculateCompleteType() {
 
   // then try the runtime
   std::vector<CompilerDecl> compiler_decls;
-  auto *objc_language_runtime = ObjCLanguageRuntime::Get(*process_sp);
-  if (auto runtime_vendor = objc_language_runtime->GetDeclVendor()) {
-    if (runtime_vendor->FindDecls(class_name, false, UINT32_MAX,
-                                  compiler_decls) > 0 &&
-        compiler_decls.size() > 0) {
-      auto *ctx = llvm::dyn_cast<TypeSystemClang>(compiler_decls[0].GetTypeSystem());
-      if (ctx) {
-        CompilerType runtime_type =
-            ctx->GetTypeForDecl(compiler_decls[0].GetOpaqueDecl());
-        m_override_type =
-          is_pointer_type ? runtime_type.GetPointerType() : runtime_type;
+  if (auto *objc_language_runtime = ObjCLanguageRuntime::Get(*process_sp)) {
+    if (auto runtime_vendor = objc_language_runtime->GetDeclVendor()) {
+      if (runtime_vendor->FindDecls(class_name, false, UINT32_MAX,
+                                    compiler_decls) > 0 &&
+          compiler_decls.size() > 0) {
```

Comment:
This may need to be fixed on llvm.org instead. If this is what is on llvm.org, please disregard this comment. (Feel free to submit a patch on reviews.llvm.org though!)

```
runtime_vendor->FindDecls(class_name, false, UINT32_MAX, compiler_decls);
if (!compiler_decls.empty()) ...
```

Reply:
Sorry Adrian, I don't understand what you mean. What part should be fixed on llvm's repo? This function is a lot different from the current llvm-project/master version, there's no Objc runtime check there.

Also, the null check I added here is currently in master-next, but seems to have been dropped accidentaly [here](https://github.com/apple/llvm-project/commit/4c278f2a2bae15233065309a6c5d307bc7caba2a), which lead me to believe we should re-add it here.

Reply:
@adrian-prantl I understand what you mean now. I re-tested the bug and lldb is crashing again, I guess something changed between when I wrote this and the rebase. Those lines you wrote indeed fix it, but as I said, the same function in llvm-project looks very different. So should I push those changes here?

Reply:
If you don't mind the extra work, fixing llvm.org would be the right thing to do. CC aprantl@apple.com and lldb-commits on the phabricator review.

Reply:
Ok, I can do that. I'll only be able to start working on it Monday though, since  I'm going to a place with no internet tomorrow.

But could you please clarify what I'm supposed to change on llvm's code base? Are we talking about the master branch of llvm-project? The call for `FindDecls` doesn't exist on the there, so I'm confused about what exactly to change.

Reply:
Yes, you would create a patch against https://github.com/llvm/llvm-project the master branch, and upload the patch to reviews.llvm.org and CC me and lldb-commits on the review.

> I'll only be able to start working on it Monday though

LLVM development is a very asynchronous affair, and we always prefer the right fix over a quick fix and it's perfectly normal for patches to take a couple of days to make it into the tree. Thanks!

Reply:
Hi @adrian-prantl. I'm sorry, but I'm still confused on what the change in llvm-project should actually be. This is the same function on llvm-project/master:

```
CompilerType ValueObject::MaybeCalculateCompleteType() {
  CompilerType compiler_type(GetCompilerTypeImpl());

  if (m_did_calculate_complete_objc_class_type) {
    if (m_override_type.IsValid())
      return m_override_type;
    else
      return compiler_type;
  }

  m_did_calculate_complete_objc_class_type = true;

  ProcessSP process_sp(
      GetUpdatePoint().GetExecutionContextRef().GetProcessSP());

  if (!process_sp)
    return compiler_type;

  if (auto *runtime =
          process_sp->GetLanguageRuntime(GetObjectRuntimeLanguage())) {
    if (llvm::Optional<CompilerType> complete_type =
            runtime->GetRuntimeType(compiler_type)) {
      m_override_type = complete_type.getValue();
      if (m_override_type.IsValid())
        return m_override_type;
    }
  }
  return compiler_type;
}
```
The block of code that deals with the `ObjCLanguageRuntime` and the runtime vendor doesn't exist. So is there something to do in llvm-project/master? If so, what would that look like?

Reply:
If it doesn't exist then we only need to update swift-lldb. It's unfortunate that there is swift-specific code in a non-Swift-specific generic source file, and we should strive to fix that, but for now, there's nothing to be done in llvm-project. Thank you!

Reply:
OK, so I'll push the change here!